EIPS/eip-4762.md

```diff
@@ -54,6 +54,14 @@ process this access event:
 
 Note: a non-value-bearing `SELFDESTRUCT` or `*CALL`, targetting a precompile or a system contract, will not cause the `BASIC_DATA_LEAF_KEY` to be added to the witness.
 
+If a `*CALL` or `SELFDESTRUCT`, `EXTCODESIZE`, `EXTCODEHASH` or `EXTCODECOPY` targetting a non-precompile, non-system contract, also process the access event:
+
+```
+(address, 0, CODEHASH_LEAF_KEY)
+```
+
+Note: this latter access is mean to check whether the contract is a [EIP-7702](./eip-7702.md) contract, or an EOF contract.
```

Comment:
So this is the weird bit, I am using the code hash to determine if it's a 7702 contract. Come to think of it, it should be reading the first code chunk directly because then you will also know what is the delegation address.

So yeah, the question is: apart from `EXTCODEHASH`, is there any reason to read the code hash for all these extra instructions?

Reply:
Yes, I believe we should ask to read the first code-chunk to check if there's a delegation. Also, it might worth clarifying (as 7702 does) that we shouldn't follow a chain of delegations.

> So yeah, the question is: apart from EXTCODEHASH, is there any reason to read the code hash for all these extra instructions?

I don't think so.